tests/load/orchestrator.go

```diff
@@ -16,298 +13,83 @@ import (
-	// does not pause issuance at the target and persistently under-issue and
-	// fail to account for the time it takes to add more load.
-	TxRateMultiplier float64
-
-	// The time period which TPS is averaged over
-	// Similarly, the time period which the orchestrator will wait before
-	// computing the average TPS.
-	SustainedTime time.Duration
-	// The number of attempts to try achieving a given target TPS before giving up.
-	MaxAttempts uint64
-
-	// Whether the orchestrator should return if the maxTPS has been reached
-	Terminate bool
-}
-
-// NewOrchestratorConfig returns a default OrchestratorConfig with pre-set parameters
-// for gradual load testing.
-func NewOrchestratorConfig() OrchestratorConfig {
-	return OrchestratorConfig{
-		MaxTPS:           5_000,
-		MinTPS:           1_000,
-		Step:             1_000,
-		TxRateMultiplier: 1.3,
-		SustainedTime:    20 * time.Second,
-		MaxAttempts:      3,
-		Terminate:        true,
-	}
-}
-
-// Orchestrator tests the network by continuously sending
-// transactions at a given rate (currTargetTPS) and increasing that rate until it detects that
-// the network can no longer make progress (i.e. the rate at the network accepts
-// transactions is less than currTargetTPS).
-type Orchestrator[T TxID] struct {
-	agents  []Agent[T]
-	tracker *Tracker[T]
-
-	log logging.Logger
-
-	maxObservedTPS atomic.Int64
-
-	observerGroup *errgroup.Group
-	issuerGroup   *errgroup.Group
-
-	config OrchestratorConfig
+type Orchestrator struct {
+	wallets  []*Sender
+	builders []Builder
+	tracker  *Tracker
+	log      logging.Logger
 }
 
-func NewOrchestrator[T TxID](
-	agents []Agent[T],
-	tracker *Tracker[T],
+func NewOrchestrator(
+	wallets []*Sender,
+	builders []Builder,
+	tracker *Tracker,
 	log logging.Logger,
-	config OrchestratorConfig,
-) *Orchestrator[T] {
-	return &Orchestrator[T]{
-		agents:  agents,
-		tracker: tracker,
-		log:     log,
-		config:  config,
+) *Orchestrator {
+	return &Orchestrator{
+		wallets:  wallets,
+		builders: builders,
+		tracker:  tracker,
+		log:      log,
 	}
 }
 
-func (o *Orchestrator[_]) Execute(ctx context.Context) error {
-	ctx, cancel := context.WithCancel(ctx)
-
-	// start a goroutine to confirm each issuer's transactions
-	o.observerGroup = &errgroup.Group{}
-	for _, agent := range o.agents {
-		o.observerGroup.Go(func() error { return agent.Listener.Listen(ctx) })
+func (g *Orchestrator) Run(ctx context.Context) error {
+	g.log.Debug("starting run")
+	issuanceF := func(i IssuanceReceipt) {
+		g.tracker.LogIssuance(i)
 	}
 
-	// start the test and block until it's done
-	success := o.run(ctx)
-
-	var err error
-	if !success {
-		err = ErrFailedToReachTargetTPS
+	confirmationF := func(c ConfirmationReceipt) {
+		g.tracker.LogConfirmation(c)
 	}
 
-	// stop the observers and issuers
-	cancel()
-
-	// block until both the observers and issuers have stopped
-	return errors.Join(o.issuerGroup.Wait(), o.observerGroup.Wait(), err)
-}
+	eg, cctx := errgroup.WithContext(ctx)
```

Comment:
To get around the weird `cctx` naming I would just name `ctx` -> `rootCtx` or `parentCtx` or call `cctx` -> `childCtx` or both